src/backend/utils.ts

```diff
@@ -942,6 +950,44 @@ async function ContinueWithFoundWine(
   return { response }
 }
 
+export async function downloadDefaultWine() {
+  // refresh wine list
+  await updateWineVersionInfos(true)
+  // get list of wines on wineDownloaderInfoStore
+  const availableWine = wineDownloaderInfoStore.get('wine-releases', [])
+  // use Wine-GE type if on Linux and Wine-Crossover if on Mac
+  const release = availableWine.filter((version) => {
+    if (isLinux) {
+      return version.version.includes('Wine-GE-Proton')
+    } else if (isMac) {
+      return version.version.includes('Wine-Crossover')
+    }
+    return false
+  })[0]
```

Comment:
Well, `.pop()` will correctly type your `release` as being potentially undefined. With `[0]` you have a condition that doesn't *look like* it can ever be true

Reply:
What do you mean? in case the list is empty this release will be `undefined`.

Reply:
you mean like this:
<img width="629" alt="image" src="https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/26871415/461d9f4d-dfec-4af0-b885-459e186ea3f9">

But then I will get the latest element instead of the first one.

Reply:
The condition can be true, yes. But the type of `release` is `WineversionInfo`, *not* `WineVersionInfo | undefined`, meaning on a type level, `!release` should always be `false`.

The difference here is that with `[0]`, you can just remove the `if (!release)` case & nothing will look broken. If you'd use `.pop()` on the other hand, you'd immediately get TypeScript yelling at you since you're trying to pass a `WineVersionInfo | undefined` to a function that accepts only `WineVersionInfo` (this function being `installWineVersion`)

Reply:
(Oh whoops, it's `shift`, not `pop`)

Reply:
I see what you mean, ok, but in the end, the result will be the same. It is just a TS thing, in the end, everything is JS :P
I also think that for anyone that will read this code in the future, using shift or the first array element will mean the same, am I wrong? bottom line, what matters in the end is that the release should exist so it will be passed along to the other method.

Reply:
You are correct in that they'll ultimately do the same thing (and you're free to use whatever you want), I'd just say that being able to spot potential issues easier is a neat bonus of one approach